src/bin/pgcopydb/summary.c

```diff
@@ -2347,21 +2322,29 @@ catalog_timing_fetch(SQLiteQuery *query)
 	timing->doneTime = sqlite3_column_int64(query->ppStmt, 3);
 	timing->durationMs = sqlite3_column_int64(query->ppStmt, 4);
 
-	if (sqlite3_column_type(query->ppStmt, 5) != SQLITE_NULL)
+	/*
+	 * Skip reading `ppDuration` from DB and create it from `durationMs`
+	 * in accordance with the single source of truth principle.
+	 */
+	if (timing->durationMs > 0)
 	{
-		strlcpy(timing->ppDuration,
-				(char *) sqlite3_column_text(query->ppStmt, 5),
-				sizeof(timing->ppDuration));
+		IntervalToString(timing->durationMs,
+						 timing->ppDuration,
+						 INTSTRING_MAX_DIGITS);
 	}
 
-	timing->count = sqlite3_column_int64(query->ppStmt, 6);
-	timing->bytes = sqlite3_column_int64(query->ppStmt, 7);
+	timing->count = sqlite3_column_int64(query->ppStmt, 5);
+	timing->bytes = sqlite3_column_int64(query->ppStmt, 6);
 
-	if (sqlite3_column_type(query->ppStmt, 8) != SQLITE_NULL)
+	/*
+	 * Skip reading `ppBytes` from DB and create it from `bytes`
+	 * in accordance with the single source of truth principle.
+	 */
```

Comment:
Same as before. What about external processes that might want to access to the telemetry?